### DISCUSSION TITLE
Creating a Matrix Rain code with Brython :)

### DISCUSSION THREAD

**Discussion by BobSquarePants** [brython-dev/brython#2473] [Q&A]
Hi everyone,

I had the crazy idea to create a **Matrix rain code** with Brython ! ( why not :P )

So far I created the grid (with HTML element `table` ) and have ready the generation of random characters

Now, I'm blocking to run ~simultaneously (or make like it does) the vertical stream of falling code...

Any ideas of what could I use to do so ?

Thanks.

**Comment by denis-migdal** [comment]
Hi,

You don't need to use an HTML table, you just need a div that will be your screen (CSS position:relative), and add your random characters with a span with a (CSS position:absolute).

Then, you just need to use transition to animate it, cf https://developer.mozilla.org/en-US/docs/Web/CSS/transition
Just set it top: 0px, then bottom: 0px and it should animate.

Else you can also take a look at : https://developer.mozilla.org/en-US/docs/Web/CSS/animation

When you want to do things more complex, you can also use https://developer.mozilla.org/en-US/docs/Web/API/Window/requestAnimationFrame

**Reply by BobSquarePants** [reply]
Thanks @denis-migdal ,

[requestAnimationFrame()](http://www.javascriptkit.com/javatutors/requestanimationframe.shtml) look promising ! 👍
